From cec728ae048a8af343cdfe3e08b85cb021473a93 Mon Sep 17 00:00:00 2001
From: "Benjamin A. Beasley" <code@musicinmybrain.net>
Date: Fri, 18 Sep 2026 06:54:50 +0100
Subject: [PATCH 06/11] Do not set COMPILE_WARNING_AS_ERROR

Considering new warnings can appear at any time due to changes in
dependency and compiler versions, this is too strict for distribution
packaging. It makes sense for upstream CI.
--- a/cmake/CMakeLists.txt
+++ b/cmake/CMakeLists.txt
@@ -1183,7 +1183,7 @@
       target_compile_definitions(${target_name} PRIVATE USE_KLEIDIAI)
     endif()
 
-    set_target_properties(${target_name} PROPERTIES COMPILE_WARNING_AS_ERROR ON)
+    set_target_properties(${target_name} PROPERTIES COMPILE_WARNING_AS_ERROR OFF)
     if (onnxruntime_USE_CUDA)
       # Suppress a "conversion_function_not_usable" warning in gsl/span
       target_compile_options(${target_name} PRIVATE "$<$<COMPILE_LANGUAGE:CUDA>:SHELL:-Xcudafe \"--diag_suppress=conversion_function_not_usable\">")
